endpoints-common/src/main/scala/com/thenewmotion/ocpi/common/AuthorizedRequests.scala

```diff
@@ -18,11 +17,26 @@ trait AuthorizedRequests {
   private val logRequest: HttpRequest => HttpRequest = { r => logger.debug(r.toString); r }
   private val logResponse: HttpResponse => HttpResponse = { r => logger.debug(r.toString); r }
 
-  protected def requestWithAuth(http: HttpExt, req: HttpRequest, auth: AuthToken[Ours])
-    (implicit ec: ExecutionContext, mat: Materializer): Future[HttpResponse] = {
-    http.singleRequest(logRequest(req.addCredentials(GenericHttpCredentials("Token", auth.value, Map())))).map { response =>
+  private def requestWithAuthSupportingRedirect(
+    http: HttpExt, req: HttpRequest, auth: AuthToken[Ours], redirectCount: Int = 0
+  )(implicit ec: ExecutionContext, mat: Materializer): Future[HttpResponse] = {
+    http.singleRequest(
+      logRequest(req.addCredentials(GenericHttpCredentials("Token", auth.value, Map())))
+    ).flatMap { response =>
       logResponse(response)
-      response
+      response.status match {
+        case StatusCodes.PermanentRedirect | StatusCodes.TemporaryRedirect if redirectCount < 10 =>
+          response.header[Location].map { newLoc =>
+            logger.warn("Following redirect to {}", newLoc.uri)
```

Comment:
not so sure about the warning here. i would go for `info`

Reply:
I thought warning because if there's a redirect then you should switch to use the new url, by following redirects you are making at least 2 requests for every one, so it's something that should be fixed